# Results

![
**Overview of the metagenome analysis technique used in this paper.** Steps that are outlined in grey were developed in this paper. **Step 1:** Using quality controlled sequencing reads from many metagenomes, we decomposed reads into k-mers and subsample these k-mers using FracMinHash, thereby selecting k-mers that evenly represent the sequence diversity within a sample. We then identified interesting k-mers using random forests, and associate these k-mers with genomes in reference databases. **Step 2:** For each metagenome, we constructed a compact de Bruijn assembly graph that contains all k-mers from a metagenome. We used dominating sets to carve the graph into pieces. We queried this graph with genomes associated with interesting k-mers identified in Step 1, recovering sequence diversity nearby in the assembly graph. We refer to these sequences as genome query neighborhoods. Step 2 is the workflow published in [@doi:10.1186/s13059-020-02066-4]. **Step 3:** We combined genome query neighborhoods for a single genome from all metagenomes. We constructed a compact de Bruijn assembly graph from these sequences, and used a dominating set with a large radius to carve the graph into large pieces. Here, we diagram construction of R=2 dominating set pieces, but in practice we used R=10. We estimated the abundance of k-mers in each metagenome for each dominating set piece, and used these abundances to perform differential abundance analysis.
](images/ibd_overview_steps.svg "Pipeline overview"){#fig:overview}

<!-- what should this first section say? -->

## K-mers are weakly predictive of IBD subtype

| Cohort      |   Name        | Country      |Total   |CD      |UC      |nonIBD  | Reference |
|:------------|:-------------:|:------------:|:------:|:------:|:------:|:------:|:----------|
| iHMP        | IBDMDB                   | USA              | 106   | 50  | 30  | 26 | [@doi:10.1038/s41586-019-1237-9] |
| PRJEB2054   | MetaHIT                  | Denmark, Spain   | 124   | 4   | 21  | 99 | [@doi:10.1038/nature08821] |
| SRP057027   | NA                       | Canada, USA      | 112   | 87  | 0   | 25 | [@doi:10.1016/j.chom.2015.09.008] |
| PRJNA385949 | PRISM, STiNKi            | USA              | 17    | 9   | 5   | 3 | [@doi:10.1186/s13073-017-0490-5] |
| PRJNA400072 | PRISM, LLDeep, and NLIBD | USA, Netherlands | 218   | 87  | 76  | 55 | [@doi:10.1038/s41564-018-0306-4] |
| PRJNA237362 | RISK                     | North America    | 28    | 23  | 0   | 5 | [@doi:10.1016/j.chom.2014.02.005] | 
| Total       |                          |                  | 605   | 260 | 132 | 213| |
Table: Six IBD shotgun metagenome sequencing cohorts used in this meta-cohort analysis.
{#tbl:cohort}

We developed a reference-free pipeline to fully characterize gut metagenomes of IBD patients (**Figure @fig:overview**).
After consistent pre-processing, we use FracMinHash sketching to produce subsampled k-mer abundance profiles of metagenomes that reflect the sequence diversity in a sample [@doi:10.12688/f1000research.19675.1] (CITE: Gather), and use these profiles to perform metagenome-wide k-mer association with IBD subtype. 
We refer to FracMinHash sketches as *signatures*, and for simplicity, continue referring to the sub-sampled k-mers in a signature as *k-mers*. 
In total, we profiled 7,376,151 subsampled k-mers across all samples in all cohorts, representing approximately 14 billion total k-mers. 
We detected variation due to IBD diagnosis in k-mer profiles of gut metagenomes from different cohorts.
We calculated pairwise distance matrices using jaccard distance and angular distance between k-mer profiles, where jaccard distance captured sample richness and angular distance captured sample diversity. 
We performed principle coordinate analysis and PERMANOVA with these distance matrices (**Figure @fig:kmers**), using the variables study accession, diagnosis, library size, and number of k-mers observed in a sample (**Table @tbl:permanova**). 
Number of k-mers observed in a sample accounted for the highest variation, possibly reflecting reduced diversity in stool metagenomes of CD and UC patients (reviewed in [@doi:10.1038/s41579-019-0213-6 ]). 
Study accounted for the second highest variation, emphasizing that technical artifacts can introduce strong signals that may influence heterogeneity in IBD microbiome studies but that can be mitigated through meta-cohort analysis [@doi:10.1038/s41591-019-0406-6].
Diagnosis accounted for a similar amount of variation as study, indicating that there is a small but detectable signal of IBD subtype in stool metagenomes.


|Variable          |Jaccard distance | Angular distance|
|:-----------------|:---------------:|:---------------:|
| Number of k-mers | 9.9%           | 6.2% |
| Study accession  | 6.6%           | 13.5% |
| Diagnosis        | 6.2%           | 3.3% |
| Library size     | 0.009%        | 0.01% |
Table: Results from PERMANOVA performed on Jaccard and Angular distance matrices. Number of k-mers refers to the number of k-mers in a signature, while library size refers to the number of raw reads per sample. All test were significant at p < .001.
{#tbl:permanova}

To evaluate whether the variation captured by diagnosis is predictive of IBD subtype, we built random forests classifiers to predict CD, UC, or nonIBD subtype.
Random forests is a supervised learning classification model that estimates how predictive k-mers are of IBD subtype, and weights individual k-mers as more or less predictive using a metric called variable importance.
To assess whether disease signatures generalize across study populations, we used a leave-one-study-out cross-validation approach where we built and optimized a classifier using five cohorts and validated on the sixth.
We built each model six times, using a separate random seed each time, to hone in on cross-study and cross-model signal.
Given the high-dimensional structure of this data set (e.g. many more k-mers than metagenomes), we first used variable selection to narrow the set of predictive k-mers in the training set [@doi:10.1007/s11634-016-0276-4; @doi:10.1093/bib/bbx124].
Variable selection reduced the number of k-mers used in each model by two orders of magnitude, from 7,376,151 to 28,684-41,701 (mean = 35,673.1, sd = 4090.3) (**fig @fig:kmers**). 
<!-- @TR: RECALCULATE REDUCTION) -->
Using this reduced set of k-mers, we then optimized each random forests classifier on the training set, producing 36 optimized models. 
We validated each model on the left-out study.
The accuracy on the validation studies ranged from 49%-77% (**Figure @fig:kmers**), outperforming a previously published model built on metagenomic data alone [@doi:10.1038/s41564-018-0306-4].

<!-- TR UPDATE TO GENOME APPROACH -->

To understand which species were responsible for disease signatures detected by our models, we anchored k-mers in the models against genomes in the GTDB rs202 representatives database using sourmash gather.
Sourmash gather determines the minimum set of genomes in database necessary to cover all of the k-mers in a query (CITE: Gather).
We found that a substantial fraction of genomes were shared between models, indicating there is a consistent biological signal captured among classifiers.
Of XX total genomes detected across all classifiers, 360 genomes were shared between all classifiers (supplemental upset figure? will be atrocious bc 36 is a lot of sets).
The presence of shared k-mers between classifiers indicates that there is a weak but consistent biological signal in metagenomes for IBD subtype between cohorts.

K-mers that anchored to these shared genomes represented XX% of all k-mers used to build the optimized classifiers, but accounted for an outsize proportion of variable importance in the optimized classifiers.
After normalizing variable importance across classifiers, XX% of the total variable importance was held by these k-mers. 
These k-mers contribute a large fraction of predictive power for classification of IBD subtype, and the genomes in which they are found represent a microbial core that contains predictive power in IBD subtype classification.

INCLUDE DETAILS OF HOW WE GOT FROM 360 GENOMES TO 54.

## nbhds

To recover strain variation in the metagenomes but not in the reference databases for these 54 genomes...

## IBD gut microbiomes have decreased diversity in strict anaerobes that is punctuated by strain switches for some facultative anaerobes

<!-- TR: This paragraph was originally written for k-mers, and now writing it for dominating sets is clunky so may need to be revised. But i like its biology tie ins -->
Many beneficial bacteria, in particular those that are oxygen sensitive, decreased in abundance in CD and UC.
Nine of the 54 species we investigated were *Faecalibacterium prausnitzii* of some sort in GTDB (**Figure **). 
<!-- TR: how to properly talk about this -- they're different species in GTDB but the same in the literature... -->
*F. prausnitzii* is an obligate anaerobe and a key butyrate producer in the gut, and plays a crucial role in reducing intestinal inflammation [@doi:10.1038/ismej.2016.176].
*F. prausnitzii* is extremely sensitive to oxygen, though it may be able to withstand oxygen exposure for up to 24 hours depending on the availability of metabolites for extracellular electron transfer [@doi:10.1038/ismej.2016.176].
In the origianl models, *Acetatifactor sp900066565* held the largest variable importance largest variable importance, and dominating set pieces decreased in abundance in CD and to a lesser extent UC (**Figure**).
*Acetatifactor* is a bile-acid producing bacteria associated with a healthy gut, but limited evidence has associated it with decreased abundance in IBD [@doi:10.1002/hep.29857].
In UC, *Gemmiger formicilis* had the most dominating set pieces decrease in abundance compared to nonIBD (**Figure**).
*G. formicilis* is a strictly anaerobic bacteria that produces both formic acid and butyric acid [@doi:10.1099/00207713-25-2-202].
We also observed a decrease in other oxygen-sensitive species, including *Lachnospira eligens*. 
*L. eligans* is an obligate anaerobe that is unable to tolerate atmospheric oxygen for an hour [@doi:10.1186/s13073-017-0490-5]. 
Collectively, the decrease in species diversity we observed in IBD, in particular CD, is consistent with a shift toward increased oxidative stress during disease that is intolerable for many gut microbes [@doi:10.1038/ismej.2013.80].

A substantial fraction of dominating set pieces were more abundant in CD than nonIBD in the metapangenome graphs of five species (**Figure**).

Only the increase in the distinct accessory genes is detectable amongst the backdrop of general loss of gene abundance due to decrease in other strains.
...
Enrichment of specific metabolic pathways is consistent with functional specialization of strains in different environmental niches [@doi:10.15252/msb.20177589].
These four genomes annotated to *Ruminococcus gnavus* ...
<!-- *Clostridium bolteae* -->

*F. gnavus* is an aerotolerant anaerobe, one clade of which was recently found to be enriched in CD [@doi:10.1186/s13073-017-0490-5], and to produce an inflammatory polysaccharide [@doi:10.1073/pnas.1904099116]. 
*C. bolteae* is a member of the normal gut microbiota but is an opportunistic pathogen that exploits compromised intestinal barriers [@doi:10.1186/s12864-016-3152-x].
It is associated with disturbance succession and has increased gene expression during gut dysbiosis [@doi:10.1101/gr.138198.112; @doi:10.1038/s41586-019-1237-9].



## Fractions of some metapangenomes are more abundant in but not exclusive to IBD

ORPHEUM/PANMERS ANALYSIS FOR STRAIN SWITCHERS

Given the signatures of strain enrichment we detected in metagenomes from CD and UC, we next investigated whether there is a disease-specific metapangenome in CD or UC -- i.e., whether there are genes from a species that are only observed in IBD.
To perform this anlysis in an assembly-free way, we used a recently developed approach that predicts open reading frames directly from short sequencing reads and then uses protein k-mers from those reads to estimate pangenomes (CITE: panmers). 
We generated standard pangenome metrics from reads from query genome neighborhoods.

In general, we found no evidence for disease-specific pangenomes among the XX pangenomes.
Instead, in almost all pangenomes all protein k-mers are observed in at least some CD, UC, and nonIBD metagenomes.
On average, XX protein k-mers were unobserved in UC, XX in CD, and XX in nonIBD per metapangenome, accounting for less than 1% of protein k-mers in the complete metapangenome (Table XX).
These results in part explain heterogeneous study findings in previous IBD gut microbiome investigations.

<!-- While we found no evidence of a general disease-specific metapangenome, we tested whether the biosynthetic cluster for the inflammatory polysaccharide produced by *F. gnavus* occurred in nonIBD as it had previously only been identified in CD [@doi:10.1073/pnas.1904099116].
An average of more than 100 reads mapped per gene in the cluster in 10 of 213 nonIBD metagenomes.
While more abundant in CD, this cluster is also identifiable within healthy human gut microbiomes, further supporting the lack of disease-specific pangenomes. -->

<!-- TR: only do this if we orpheum/panmer everything? -->
<!-- While the majority of genes were observed in CD, UC, and nonIBD, IBD metagenomes harbored fewer protein k-mers than nonIBD metagenomes (Table XX).
When we counted the number of protein k-mers observed within each metagenome, XX of 54 metapangenomes for CD and XX of 54 metapangenomes for UC had a significantly lower number of genes observed per metagenome than nonIBD (ANOVA p < 0.05, Tukey's HSD p < 0.05). 
Only *C. bolteae* in CD and *Romboutsia timonensis*, *Anaeromassilibacillus*, and *Actulibacter* in UC had a significantly more genes per sample within each pangenome.
Only *F. gnavus* showed no significant difference in the number of genes per sample across all groups. -->