# Supplementary information


![
**Fifty-four genomes are important across models and anchor the majority of variable importance.**.
The bottom panel depicts a heat map of the scale variable importance contributed by k-mers that anchored to each of the top 54 genomes that were important for predicting IBD subtype. 
Models are labelled by the validation study and by the random seed used to build the model.Rank corresponds to the genome that anchored the most variable importance. 
Rank:species can be decoded using the tree in the main manuscript. The top panels depict bar charts that correspond to the minimum (lower) or maximum (upper) variable importance a genome could anchor. 
The minimum variable importance was estimated following the sourmash gather algorithm, where each important k-mer was assigned to only one genome, and the genome it was assigned to was determined by a greedy winner-takes-all approach. 
Therefore, in the minimum bar chart, variable importance attributable to a k-mer was only summed once per k-mer, even if that k-mer occurred in multiple genomes. 
The maximum variable importance was estimated by allowing k-mers to be anchored to multiple genomes, so all k-mers were assigned to all possible genomes even if that meant a k-mer was assigned multiple times.
](images/varimp54.png){#fig:varimp}

![
**Most differentially abundant sequences occur in metagenomes of individuals diagnosed with CD, UC and non-IBD.** Upset plot of k-mers that were decreased in abundance in CD and their occurrence in CD, UC, and nonIBD metagenomes.
](images/dda_kmers_shared_across_diagnosis_upset_decreased_supplemental.png){#fig:exclusive2}
